### PR TITLE
Add purl2url support for rust

### PR DESCRIPTION
**Pull request by TG1999**
Solves issue #13 
Signed-off-by: TG1999 <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
Got this @majurg

**Comment by TG1999** [comment]
Done @majurg

File: purl2url/rust.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+def get_rust_url(purl):
+    """
+    Return a downloadable normalized URL for a given rust based `purl`
+    """
+    url_parts = urlparse(purl)
+    scheme = url_parts.scheme
+    if not(scheme == 'pkg'):
+        raise Exception('Not a valid PURL') 
+
```
Comment (majurg) [inline purl2url/rust.py]: Remove this empty space.

File: purl2url/rust.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+from urllib.parse import urlparse
+
+
+def get_rust_url(purl):
+    """
+    Return a downloadable normalized URL for a given rust based `purl`
+    """
+    url_parts = urlparse(purl)
+    scheme = url_parts.scheme
+    if not(scheme == 'pkg'):
```
Comment (majurg) [inline purl2url/rust.py]: change this to `if scheme != 'pkg':`

Comment (TG1999) [reply purl2url/rust.py]: Got this

File: purl2url/rust.py (hunk 1 above)
Comment (TG1999) [reply purl2url/rust.py]: Got this

**Comment by TG1999** [comment]
Done @majurg

File: purl2url/rust.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def get_rust_url(purl):
+    """
+    Return a downloadable normalized URL for a given rust based `purl`
+    """
+    url_parts = urlparse(purl)
+    scheme = url_parts.scheme
+    if scheme != 'pkg':
+        raise Exception('Not a valid PURL')
```
Comment (majurg) [inline purl2url/rust.py]: We should add a bit more detail here. We want to know why the given thing is not a valid pURL. What was missing from the string? The user would want to know this.

File: purl2url/rust.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        raise Exception('Not a valid PURL') 
+
+    else :
+        path = url_parts.path
+        part_paths = path.split('@')
+        version = part_paths[1]
+
+        part_paths = part_paths[0].split('/')
+        if 'crates' != part_paths[0]:
+            raise Exception('Not a valid crate PURL')
```
Comment (majurg) [inline purl2url/rust.py]: Same thing as above. Why is the given pURL not a valid crate pURL?

Comment (TG1999) [reply purl2url/rust.py]: Got this one too

**Comment by TG1999** [comment]
Done @majurg , BTW I think we should wait for https://github.com/package-url/packageurl-python/pull/24 to get merged first, because we don't know which type it should be 'crates' or 'cargo', what say?

**Comment by TG1999** [comment]
@majurg good to go ?

File: purl2url/rust.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+def get_rust_url(purl):
+    """
+    Return a downloadable normalized URL for a given rust based `purl`
+    """
+    url_parts = urlparse(purl)
+    scheme = url_parts.scheme
+    if scheme != 'pkg':
+        raise Exception('Not a valid PURL, `scheme` of URL should be `pkg` to be it a PURL for more info refer https://github.com/package-url/purl-spec/blob/master/README.rst') 
+
+    else :
```
Comment (majurg) [inline purl2url/rust.py:29]: Do we really need this else here?

Comment (TG1999) [reply purl2url/rust.py:29]: Okay got this one 👌

**Comment by TG1999** [comment]
Done @majurg

**Pull request closed without merging**